.github/workflows/build-macos-installer.yml

```diff
@@ -0,0 +1,98 @@
+name: Build macOS Installer
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
```

Comment:
This would also need a <version> inpt or something for manual dispatch, or else the manual run wouldn't know which release to upload the installer to, or what to checkout to build. Needs a followup later on the the script in checkout and upload too if you want to keep the manual dispatch.